### ISSUE TITLE
pre-commit doesn't capture modules that cannot be imported

### ISSUE THREAD

**Issue by yaelbh** [Qiskit/ibm-quantum-schemas#95]
I have this line in my code, note the typo in `schmeas`:
```python
from ibm_quantum_schmeas.aliases import Self
```

The  typo was not detected by `pre-commit`, and I noticed it only when tests failed.

**Comment by yaelbh** [comment]
Copilot says that ruff (unlike pylint) does not check if modules can be imported, and suggests to use mypy.
